src/Symfony/Component/HttpClient/Response/CurlResponse.php

```diff
@@ -63,18 +63,18 @@ public function __construct(CurlClientState $multi, $ch, array $options = null,
         }
 
         if (null === $content = &$this->content) {
-            $content = ($options['buffer'] ?? true) ? fopen('php://temp', 'w+') : null;
+            $content = isset($options['buffer']) && true === $options['buffer'] ? fopen('php://temp', 'w+') : null;
```

Comment:
`($options['buffer'] ?? true) && !\is_string($options['buffer'] ?? true)` (same in other places)

Reply:
Changed the check on all places